serverlessgo/main.go

```diff
@@ -77,9 +109,20 @@ func (lb *loadBalancer) nextInstance() *instance {
 			break
 		}
 	}
-	
 	if selected == nil {
-		selected = newInstance(len(lb.instances), lb.idlenessDeadline)
+		var err error
+		for i := 0; i < len(lb.inputs); i++ {
```

Comment:
Esse for está bem confuso. Por que vocẽ quer repetir a operação input vezes?

Sua reflexão tem que ser: O que significa o newInstance dar errado? Olhando o código você tem a resposta: o buildEntry deu errado. Se o buildEntry deu errado, você tem que matar tudo, bixo e não tentar o próximo. Ou seja, nextInstance também deve retornar um erro.

Caso você não queria fazer essa mudança, seu approach deve ser criar todos os entry arrays em tempo de inicialização. A partir dessa mudança, o newInstance nunca da erro e você não precisa se preocupar.

Reply:
Done